mesonbuild/dependencies/python.py

```diff
@@ -327,6 +327,13 @@ def __init__(self, name: str, environment: 'Environment',
         if not self.link_libpython and mesonlib.version_compare(self.version, '< 3.8'):
             self.link_args = []
 
+        # But not Apple, because it's a framework
+        if self.platform.startswith('macosx'):
+            framework_prefix = self.variables.get('PYTHONFRAMEWORKPREFIX', '')
+            has_rpath = any('rpath' in flag for flag in self.link_args)
+            # Attach rpath if needed
+            if framework_prefix.startswith('/Applications/Xcode.app/') and not has_rpath:
+                self.link_args += ['-rpath', framework_prefix]
```

Comment:
```suggestion
        if self.env.machines.host.is_darwin() and 'PYTHONFRAMEWORKPREFIX' in self.variables:
            framework_prefix = self.variables['PYTHONFRAMEWORKPREFIX']
            # Add rpath, will be de-duplicated if necessary
            if framework_prefix.startswith('/Applications/Xcode.app/'):
                self.link_args += ['-rpath,' + framework_prefix]
```

[We de-duplicate rpath arguments of the form `-Wl,-rpath,FOO`](https://github.com/mesonbuild/meson/pull/13963) so we can add it unconditionally. It's also better this way, in case some other `rpath` argument gets added to the pc file in the future, which will break projects.